source/common/stream_info/utility.h

```diff
@@ -15,22 +15,33 @@ namespace StreamInfo {
 class CustomResponseFlag {
 public:
   CustomResponseFlag(absl::string_view flag, absl::string_view flag_long);
-  uint16_t flag() const { return flag_; }
+  ExtendedResponseFlag flag() const { return flag_; }
 
 private:
-  uint16_t flag_{};
+  ExtendedResponseFlag flag_;
 };
 
 // Register a custom response flag by specifying the flag and the long name of the flag.
-// This macro should never be used in header files.
+// This macro should only be used in source files to register a flag.
 #define REGISTER_CUSTOM_RESPONSE_FLAG(short, long)                                                 \
```

Comment:
One last thing: perhaps these should have more descriptive names instead of `short` and `long`. Even just `short_name` and `long_name` would be better.